### ISSUE TITLE
Index error when rendering with pythreejs

### ISSUE THREAD

**Issue by whophil** [pyvista/pyvista#2372]
**Describe the bug, what's wrong, and what you expect:**

I encounter an index error when trying to render a certain PolyData with the pythreejs backend. Maybe some sort of off-by-one?

**To Reproduce**

```py
import pyvista as pv
pv.set_jupyter_backend('pythreejs')

mesh = pv.read('bad.vtp')
plotter = pv.Plotter()
plotter.add_mesh(mesh)
plotter.export_html('test.html')
```

[bad.zip](https://github.com/pyvista/pyvista/files/8338743/bad.zip)

Which produces the following error on my machine:
```
  File "/home/phil/miniconda3/envs/forma_devel/lib/python3.9/site-packages/pyvista/jupyter/pv_pythreejs.py", line 486, in convert_renderer
    children = meshes_from_actors(pv_renderer.actors.values(),
  File "/home/phil/miniconda3/envs/forma_devel/lib/python3.9/site-packages/pyvista/jupyter/pv_pythreejs.py", line 473, in meshes_from_actors
    mesh = actor_to_mesh(actor, focal_point)
  File "/home/phil/miniconda3/envs/forma_devel/lib/python3.9/site-packages/pyvista/jupyter/pv_pythreejs.py", line 453, in actor_to_mesh
    mesh = to_edge_mesh(dataset, mapper, prop, use_edge_coloring=False,
  File "/home/phil/miniconda3/envs/forma_devel/lib/python3.9/site-packages/pyvista/jupyter/pv_pythreejs.py", line 309, in to_edge_mesh
    edges = segment_poly_cells(surf)
  File "/home/phil/miniconda3/envs/forma_devel/lib/python3.9/site-packages/pyvista/jupyter/pv_pythreejs.py", line 40, in segment_poly_cells
    nn = cc[offset]
IndexError: index 501 is out of bounds for axis 0 with size 501
```

-----

**System Information:**

```py
>>> print(pv.Report())

--------------------------------------------------------------------------------
  Date: Wed Mar 23 23:48:28 2022 MDT

                OS : Linux
            CPU(s) : 16
           Machine : x86_64
      Architecture : 64bit
       Environment : Python
        GPU Vendor : NVIDIA Corporation
      GPU Renderer : Quadro T2000 with Max-Q Design/PCIe/SSE2
       GPU Version : 4.5.0 NVIDIA 510.54

  Python 3.9.10 | packaged by conda-forge | (main, Feb  1 2022, 21:24:11)
  [GCC 9.4.0]

           pyvista : 0.33.2
               vtk : 9.0.1
             numpy : 1.22.2
           imageio : 2.9.0
           appdirs : 1.4.4
            scooby : 0.5.7
        matplotlib : 3.5.1
             PyQt5 : 5.12.3
           IPython : 7.30.1
            meshio : 4.4.6
--------------------------------------------------------------------------------
```

**Comment by akaszynski** [comment]
Resolved with #2380

**Comment by vyas-pranav** [comment]
I found an error in the code. In line 533 for file pv_pythreejs.py, inside function convert_plotter.
